core/logger/zap.go

```diff
@@ -122,7 +122,7 @@ func (l *zapLogger) SetLogLevel(lvl zapcore.Level) {
 func (l *zapLogger) With(args ...interface{}) Logger {
 	newLogger := *l
 	newLogger.SugaredLogger = l.SugaredLogger.With(args...)
-	newLogger.fields = copyFields(l.fields, args...)
+	newLogger.fields = copyFields(newLogger.fields, args...)
```

Comment:
Could you explain what is going on here? This reads to me like it would behave identically (`newLogger.fields` was just set from `l.fields`), and IMHO it was more readable the other way since it was not overwriting itself.